packages/mesa.rb

```diff
@@ -1,373 +1,125 @@
 require 'package'
 
-class Mesa < Package
-  description 'Open-source implementation of the OpenGL specification'
-  homepage 'https://www.mesa3d.org'
-  # We use mesa amber (derived from the 21.3 series) for older kernels
-  # and current mesa versions for newer kernels.
-  if CREW_KERNEL_VERSION.to_f < 5.10
-    # Built off of the mesa amber branch
-    git_hashtag 'acfef002a081f36e6eebc6e8ab908a36ab18f68c'
-    @_ver = git_hashtag[0, 7]
-    version "amber-#{@_ver}"
-  else
+# We use mesa amber (derived from the 21.3 series) for intel hardware with older kernels
+# and current mesa versions for newer kernels.
+if !CREW_IN_CONTAINER && CREW_IS_INTEL && ( Gem::Version.new(CREW_KERNEL_VERSION) < Gem::Version.new('4.16') )
+  Package.load_package( File.join(CREW_PACKAGES_PATH, 'mesa_amber.rb') )
+  class Mesa < Package::Mesa_amber
+    def self.preflight
+      warn 'Installing mesa_amber as kernel is older than 4.16'.lightcyan
+    end
+  end
+else
```

Comment:
This looks very hackish.  Would it make sense to add this logic to preflight instead and suggest to install the other package with `crew install mesa|mesa_amber`?

Reply:
> Also, what do we do about all the packages that have `depends_on 'mesa'`? This logic would have to be applied there also, correct?

Yep

> This looks very hackish. Would it make sense to add this logic to preflight instead and suggest to install the other package with `crew install mesa|mesa_amber`?

Another way could be turning `mesa.rb` into a fake package and break down into `mesa_latest.rb` and `mesa_amber.rb` (it seems a better approach :) )

Reply:
With Mesa 23.0.x coming out this week, would it be a good time to make this sort of change now?